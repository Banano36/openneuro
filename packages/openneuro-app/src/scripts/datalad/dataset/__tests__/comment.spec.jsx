import React from 'react'
import { mount } from 'enzyme'
import Comment from '../comment.jsx'

const emptyState =
  '{"blocks":[{"key":"3sm42","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}'

describe('Comment component', () => {
  it('renders with an empty comment', () => {
    const wrapper = mount(
      <Comment
        data={{
          id: '9001',
          text: emptyState,
          user: { id: '1234', email: 'example@example.com' },
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
