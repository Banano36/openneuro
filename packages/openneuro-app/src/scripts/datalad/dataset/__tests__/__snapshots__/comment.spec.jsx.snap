// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders with an empty comment 1`] = `
<Comment
  data={
    Object {
      "id": "9001",
      "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
      "user": Object {
        "email": "example@example.com",
        "id": "1234",
      },
    }
  }
>
  <div
    className="comment"
  >
    <div
      className="row comment-header"
    >
      By example@example.com - about NaN years ago
    </div>
    <div
      className="row comment-body"
    >
      <img
        className="comment-avatar"
        src="https://www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?s=200&d=identicon"
      />
      <DraftEditor
        blockRenderMap={
          Immutable.Map {
            "ordered-list-item": Object {
              "element": "li",
              "wrapper": <ol
                className="public-DraftStyleDefault-ol"
              />,
            },
            "header-six": Object {
              "element": "h6",
            },
            "header-four": Object {
              "element": "h4",
            },
            "header-one": Object {
              "element": "h1",
            },
            "unordered-list-item": Object {
              "element": "li",
              "wrapper": <ul
                className="public-DraftStyleDefault-ul"
              />,
            },
            "atomic": Object {
              "element": "figure",
            },
            "unstyled": Object {
              "aliasedElements": Array [
                "p",
              ],
              "element": "div",
            },
            "header-two": Object {
              "element": "h2",
            },
            "code-block": Object {
              "element": "pre",
              "wrapper": <pre
                className="public-DraftStyleDefault-pre"
              />,
            },
            "blockquote": Object {
              "element": "blockquote",
            },
            "header-five": Object {
              "element": "h5",
            },
            "header-three": Object {
              "element": "h3",
            },
          }
        }
        blockRendererFn={[Function]}
        blockStyleFn={[Function]}
        editorKey="9001"
        editorState={
          EditorState {
            "_immutable": Immutable.Record {
              "allowUndo": true,
              "currentContent": Immutable.Record {
                "entityMap": Object {},
                "blockMap": Immutable.OrderedMap {
                  "3sm42": Immutable.Record {
                    "key": "3sm42",
                    "type": "unstyled",
                    "text": "",
                    "characterList": Immutable.List [],
                    "depth": 0,
                    "data": Immutable.Map {},
                  },
                },
                "selectionBefore": Immutable.Record {
                  "anchorKey": "3sm42",
                  "anchorOffset": 0,
                  "focusKey": "3sm42",
                  "focusOffset": 0,
                  "isBackward": false,
                  "hasFocus": false,
                },
                "selectionAfter": Immutable.Record {
                  "anchorKey": "3sm42",
                  "anchorOffset": 0,
                  "focusKey": "3sm42",
                  "focusOffset": 0,
                  "isBackward": false,
                  "hasFocus": false,
                },
              },
              "decorator": null,
              "directionMap": Immutable.OrderedMap {
                "3sm42": "LTR",
              },
              "forceSelection": false,
              "inCompositionMode": false,
              "inlineStyleOverride": null,
              "lastChangeType": null,
              "nativelyRenderedContent": null,
              "redoStack": Immutable.Stack [],
              "selection": Immutable.Record {
                "anchorKey": "3sm42",
                "anchorOffset": 0,
                "focusKey": "3sm42",
                "focusOffset": 0,
                "isBackward": false,
                "hasFocus": false,
              },
              "treeMap": Immutable.OrderedMap {
                "3sm42": Immutable.List [
                  Immutable.Record {
                    "start": 0,
                    "end": 0,
                    "decoratorKey": null,
                    "leaves": Immutable.List [
                      Immutable.Record {
                        "start": 0,
                        "end": 0,
                      },
                    ],
                  },
                ],
              },
              "undoStack": Immutable.Stack [],
            },
          }
        }
        keyBindingFn={[Function]}
        onChange={[Function]}
        readOnly={false}
        spellCheck={false}
        stripPastedStyles={false}
      >
        <div
          className="DraftEditor-root"
        >
          <div
            className="DraftEditor-editorContainer"
          >
            <div
              aria-describedby="placeholder-9001"
              aria-expanded={null}
              className="notranslate public-DraftEditor-content"
              contentEditable={true}
              onBeforeInput={[Function]}
              onBlur={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onCopy={[Function]}
              onCut={[Function]}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              onPaste={[Function]}
              onSelect={[Function]}
              role="textbox"
              spellCheck={false}
              style={
                Object {
                  "WebkitUserSelect": "text",
                  "outline": "none",
                  "userSelect": "text",
                  "whiteSpace": "pre-wrap",
                  "wordWrap": "break-word",
                }
              }
              suppressContentEditableWarning={true}
            >
              <DraftEditorContents
                blockRenderMap={
                  Immutable.Map {
                    "ordered-list-item": Object {
                      "element": "li",
                      "wrapper": <ol
                        className="public-DraftStyleDefault-ol"
                      />,
                    },
                    "header-six": Object {
                      "element": "h6",
                    },
                    "header-four": Object {
                      "element": "h4",
                    },
                    "header-one": Object {
                      "element": "h1",
                    },
                    "unordered-list-item": Object {
                      "element": "li",
                      "wrapper": <ul
                        className="public-DraftStyleDefault-ul"
                      />,
                    },
                    "atomic": Object {
                      "element": "figure",
                    },
                    "unstyled": Object {
                      "aliasedElements": Array [
                        "p",
                      ],
                      "element": "div",
                    },
                    "header-two": Object {
                      "element": "h2",
                    },
                    "code-block": Object {
                      "element": "pre",
                      "wrapper": <pre
                        className="public-DraftStyleDefault-pre"
                      />,
                    },
                    "blockquote": Object {
                      "element": "blockquote",
                    },
                    "header-five": Object {
                      "element": "h5",
                    },
                    "header-three": Object {
                      "element": "h3",
                    },
                  }
                }
                blockRendererFn={[Function]}
                blockStyleFn={[Function]}
                customStyleMap={
                  Object {
                    "BOLD": Object {
                      "fontWeight": "bold",
                    },
                    "CODE": Object {
                      "fontFamily": "monospace",
                      "wordWrap": "break-word",
                    },
                    "ITALIC": Object {
                      "fontStyle": "italic",
                    },
                    "STRIKETHROUGH": Object {
                      "textDecoration": "line-through",
                    },
                    "UNDERLINE": Object {
                      "textDecoration": "underline",
                    },
                  }
                }
                editorKey="9001"
                editorState={
                  EditorState {
                    "_immutable": Immutable.Record {
                      "allowUndo": true,
                      "currentContent": Immutable.Record {
                        "entityMap": Object {},
                        "blockMap": Immutable.OrderedMap {
                          "3sm42": Immutable.Record {
                            "key": "3sm42",
                            "type": "unstyled",
                            "text": "",
                            "characterList": Immutable.List [],
                            "depth": 0,
                            "data": Immutable.Map {},
                          },
                        },
                        "selectionBefore": Immutable.Record {
                          "anchorKey": "3sm42",
                          "anchorOffset": 0,
                          "focusKey": "3sm42",
                          "focusOffset": 0,
                          "isBackward": false,
                          "hasFocus": false,
                        },
                        "selectionAfter": Immutable.Record {
                          "anchorKey": "3sm42",
                          "anchorOffset": 0,
                          "focusKey": "3sm42",
                          "focusOffset": 0,
                          "isBackward": false,
                          "hasFocus": false,
                        },
                      },
                      "decorator": null,
                      "directionMap": Immutable.OrderedMap {
                        "3sm42": "LTR",
                      },
                      "forceSelection": false,
                      "inCompositionMode": false,
                      "inlineStyleOverride": null,
                      "lastChangeType": null,
                      "nativelyRenderedContent": null,
                      "redoStack": Immutable.Stack [],
                      "selection": Immutable.Record {
                        "anchorKey": "3sm42",
                        "anchorOffset": 0,
                        "focusKey": "3sm42",
                        "focusOffset": 0,
                        "isBackward": false,
                        "hasFocus": false,
                      },
                      "treeMap": Immutable.OrderedMap {
                        "3sm42": Immutable.List [
                          Immutable.Record {
                            "start": 0,
                            "end": 0,
                            "decoratorKey": null,
                            "leaves": Immutable.List [
                              Immutable.Record {
                                "start": 0,
                                "end": 0,
                              },
                            ],
                          },
                        ],
                      },
                      "undoStack": Immutable.Stack [],
                    },
                  }
                }
                key="contents0"
              >
                <div
                  data-contents="true"
                >
                  <div
                    className=""
                    data-block={true}
                    data-editor="9001"
                    data-offset-key="3sm42-0-0"
                    key="3sm42"
                  >
                    <DraftEditorBlock
                      block={
                        Immutable.Record {
                          "key": "3sm42",
                          "type": "unstyled",
                          "text": "",
                          "characterList": Immutable.List [],
                          "depth": 0,
                          "data": Immutable.Map {},
                        }
                      }
                      blockStyleFn={[Function]}
                      contentState={
                        Immutable.Record {
                          "entityMap": Object {},
                          "blockMap": Immutable.OrderedMap {
                            "3sm42": Immutable.Record {
                              "key": "3sm42",
                              "type": "unstyled",
                              "text": "",
                              "characterList": Immutable.List [],
                              "depth": 0,
                              "data": Immutable.Map {},
                            },
                          },
                          "selectionBefore": Immutable.Record {
                            "anchorKey": "3sm42",
                            "anchorOffset": 0,
                            "focusKey": "3sm42",
                            "focusOffset": 0,
                            "isBackward": false,
                            "hasFocus": false,
                          },
                          "selectionAfter": Immutable.Record {
                            "anchorKey": "3sm42",
                            "anchorOffset": 0,
                            "focusKey": "3sm42",
                            "focusOffset": 0,
                            "isBackward": false,
                            "hasFocus": false,
                          },
                        }
                      }
                      customStyleMap={
                        Object {
                          "BOLD": Object {
                            "fontWeight": "bold",
                          },
                          "CODE": Object {
                            "fontFamily": "monospace",
                            "wordWrap": "break-word",
                          },
                          "ITALIC": Object {
                            "fontStyle": "italic",
                          },
                          "STRIKETHROUGH": Object {
                            "textDecoration": "line-through",
                          },
                          "UNDERLINE": Object {
                            "textDecoration": "underline",
                          },
                        }
                      }
                      decorator={null}
                      direction="LTR"
                      forceSelection={false}
                      key="3sm42"
                      offsetKey="3sm42-0-0"
                      selection={
                        Immutable.Record {
                          "anchorKey": "3sm42",
                          "anchorOffset": 0,
                          "focusKey": "3sm42",
                          "focusOffset": 0,
                          "isBackward": false,
                          "hasFocus": false,
                        }
                      }
                      tree={
                        Immutable.List [
                          Immutable.Record {
                            "start": 0,
                            "end": 0,
                            "decoratorKey": null,
                            "leaves": Immutable.List [
                              Immutable.Record {
                                "start": 0,
                                "end": 0,
                              },
                            ],
                          },
                        ]
                      }
                    >
                      <div
                        className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                        data-offset-key="3sm42-0-0"
                      >
                        <DraftEditorLeaf
                          block={
                            Immutable.Record {
                              "key": "3sm42",
                              "type": "unstyled",
                              "text": "",
                              "characterList": Immutable.List [],
                              "depth": 0,
                              "data": Immutable.Map {},
                            }
                          }
                          customStyleMap={
                            Object {
                              "BOLD": Object {
                                "fontWeight": "bold",
                              },
                              "CODE": Object {
                                "fontFamily": "monospace",
                                "wordWrap": "break-word",
                              },
                              "ITALIC": Object {
                                "fontStyle": "italic",
                              },
                              "STRIKETHROUGH": Object {
                                "textDecoration": "line-through",
                              },
                              "UNDERLINE": Object {
                                "textDecoration": "underline",
                              },
                            }
                          }
                          forceSelection={false}
                          isLast={true}
                          key="3sm42-0-0"
                          offsetKey="3sm42-0-0"
                          selection={
                            Immutable.Record {
                              "anchorKey": "3sm42",
                              "anchorOffset": 0,
                              "focusKey": "3sm42",
                              "focusOffset": 0,
                              "isBackward": false,
                              "hasFocus": false,
                            }
                          }
                          start={0}
                          styleSet={Immutable.OrderedSet []}
                          text=""
                        >
                          <span
                            data-offset-key="3sm42-0-0"
                            style={Object {}}
                          >
                            <DraftEditorTextNode>
                              <br
                                data-text="true"
                                key="B"
                              />
                            </DraftEditorTextNode>
                          </span>
                        </DraftEditorLeaf>
                      </div>
                    </DraftEditorBlock>
                  </div>
                </div>
              </DraftEditorContents>
            </div>
          </div>
        </div>
      </DraftEditor>
    </div>
    <LoggedIn />
  </div>
  <div
    className="row replies"
  >
    <div
      className="comment-reply"
    />
  </div>
</Comment>
`;
